Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -0,0 +1,40 @@
+# Proposal for meta/run.yaml content being present in the composed image
+
+Currently, Capstan runtime support makes use of `$CWD/meta/run.yaml` for composing and running an OSv image. Although this is fine for the composing, it gets tricky for running because `capstan run <IMAGE>` generally does not require user to be inside package root directory.
+
+I propose the following solution:
+
+### Composing image
+File meta/run.yaml is present at package root directory (same as until now)
+
+When user runs `capstan package compose` the content of meta/run.yaml file is processed and new files are generated inside the OSv image folder `/osv-runtime`:
+
+Capstan should create a separate file for each named configuration e.g.
+```
+/osv-runtime
+   |--burek
+   |--pizza   <--------\
+   |                    |
+   |--default-runtime  /
+```
+Each runtime file should contain only one line representing the run command that is to be run inside OSv:
+```
+# content of file /osv-runtime/burek:
+node burek.js
+```
+
+When user runs `capstan package compose` the boot command is set so that the agent is run. A capstan package must be implemented that provides agent funcionality.
+
+#### OSv Agent
+Agent reads *cloud-init* file. Based on cloud-init data it spawns a thread with appropriate run command.
+
+TODO: Agent should be able to do more than just pick runtime! Since cloud-init script could contain some runtime-specific settings, say master-IP, and agent should be able to use them. At least it should be able to replace placeholders within run command. Must be careful, though, that agent needn't be adjusted when new runtime is added!
```

Comment:
This is what I had in mind before: https://cloudinit.readthedocs.io/en/latest/topics/examples.html#run-commands-on-first-boot

OSv does not have this supported yet, but it's trivial to add. Then it would be quite trivial to add `runcmd` and request `agent run <name-of-config>`. This would invoke the agent that would lookg for the config and run it.

Eventually, the cloud-init config will be generated by some other means allowing users to just compose their apps not really having to know how to launch the app, just its name.